// @storybook/core/dist/preview-errors.js
var oe = Object.defineProperty;
var o = (c, t) => oe(c, "name", { value: t, configurable: !0 });

// ../node_modules/ts-dedent/esm/index.js
function n(c) {
  for (var t = [], e = 1; e < arguments.length; e++)
    t[e - 1] = arguments[e];
  var a = Array.from(typeof c == "string" ? [c] : c);
  a[a.length - 1] = a[a.length - 1].replace(/\r?\n([\t ]*)$/, "");
  var g = a.reduce(function(i, h) {
    var p = h.match(/\n([\t ]+|(?!\s).)/g);
    return p ? i.concat(p.map(function(E) {
      var d, l;
      return (l = (d = E.match(/[\t ]/g)) === null || d === void 0 ? void 0 : d.length) !== null && l !== void 0 ? l : 0;
    })) : i;
  }, []);
  if (g.length) {
    var u = new RegExp(`
[	 ]{` + Math.min.apply(Math, g) + "}", "g");
    a = a.map(function(i) {
      return i.replace(u, `
`);
    });
  }
  a[0] = a[0].replace(/^\r?\n/, "");
  var y = a[0];
  return t.forEach(function(i, h) {
    var p = y.match(/(?:^|\n)( *)$/), E = p ? p[1] : "", d = i;
    typeof i == "string" && i.includes(`
`) && (d = String(i).split(`
`).map(function(l, te) {
      return te === 0 ? l : "" + E + l;
    }).join(`
`)), y += d + a[h + 1];
  }), y;
}
o(n, "dedent");

// src/storybook-error.ts
function O({
  code: c,
  category: t
}) {
  let e = String(c).padStart(4, "0");
  return `SB_${t}_${e}`;
}
o(O, "parseErrorCode");
var m = class m extends Error {
  /**
   * Category of the error. Used to classify the type of error, e.g., 'PREVIEW_API'.
   */
  category;
  /**
   * Code representing the error. Used to uniquely identify the error, e.g., 1.
   */
  code;
  /**
   * Data associated with the error. Used to provide additional information in the error message or to be passed to telemetry.
   */
  data = {};
  /**
   * Specifies the documentation for the error.
   * - If `true`, links to a documentation page on the Storybook website (make sure it exists before enabling) – This is not implemented yet.
   * - If a string, uses the provided URL for documentation (external or FAQ links).
   * - If `false` (default), no documentation link is added.
   */
  documentation;
  /**
   * Flag used to easily determine if the error originates from Storybook.
   */
  fromStorybook = !0;
  get fullErrorCode() {
    return O({ code: this.code, category: this.category });
  }
  /**
   * Overrides the default `Error.name` property in the format: SB_<CATEGORY>_<CODE>.
   */
  get name() {
    let t = this.constructor.name;
    return `${this.fullErrorCode} (${t})`;
  }
  constructor(t) {
    super(m.getFullMessage(t)), this.category = t.category, this.documentation = t.documentation ?? !1, this.code = t.code;
  }
  /**
   * Generates the error message along with additional documentation link (if applicable).
   */
  static getFullMessage({
    documentation: t,
    code: e,
    category: a,
    message: g
  }) {
    let u;
    return t === !0 ? u = `https://storybook.js.org/error/${O({ code: e, category: a })}` : typeof t == "string" ? u = t : Array.isArray(t) &&
    (u = `
${t.map((y) => `	- ${y}`).join(`
`)}`), `${g}${u != null ? `

More info: ${u}
` : ""}`;
  }
};
o(m, "StorybookError");
var s = m;

// src/preview-errors.ts
var se = /* @__PURE__ */ ((r) => (r.BLOCKS = "BLOCKS", r.DOCS_TOOLS = "DOCS-TOOLS", r.PREVIEW_CLIENT_LOGGER = "PREVIEW_CLIENT-LOGGER", r.PREVIEW_CHANNELS =
"PREVIEW_CHANNELS", r.PREVIEW_CORE_EVENTS = "PREVIEW_CORE-EVENTS", r.PREVIEW_INSTRUMENTER = "PREVIEW_INSTRUMENTER", r.PREVIEW_API = "PREVIEW\
_API", r.PREVIEW_REACT_DOM_SHIM = "PREVIEW_REACT-DOM-SHIM", r.PREVIEW_ROUTER = "PREVIEW_ROUTER", r.PREVIEW_THEMING = "PREVIEW_THEMING", r.RENDERER_HTML =
"RENDERER_HTML", r.RENDERER_PREACT = "RENDERER_PREACT", r.RENDERER_REACT = "RENDERER_REACT", r.RENDERER_SERVER = "RENDERER_SERVER", r.RENDERER_SVELTE =
"RENDERER_SVELTE", r.RENDERER_VUE = "RENDERER_VUE", r.RENDERER_VUE3 = "RENDERER_VUE3", r.RENDERER_WEB_COMPONENTS = "RENDERER_WEB-COMPONENTS",
r.FRAMEWORK_NEXTJS = "FRAMEWORK_NEXTJS", r))(se || {}), f = class f extends s {
  constructor(e) {
    super({
      category: "PREVIEW_API",
      code: 1,
      message: n`
        Couldn't find story matching id '${e.storyId}' after HMR.
        - Did you just rename a story?
        - Did you remove it from your CSF file?
        - Are you sure a story with the id '${e.storyId}' exists?
        - Please check the values in the stories field of your main.js config and see if they would match your CSF File.
        - Also check the browser console and terminal for potential error messages.`
    });
    this.data = e;
  }
};
o(f, "MissingStoryAfterHmrError");
var D = f, R = class R extends s {
  constructor(e) {
    super({
      category: "PREVIEW_API",
      code: 2,
      documentation: "https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#using-implicit-actions-during-rendering-is-deprecated-\
for-example-in-the-play-function",
      message: n`
        We detected that you use an implicit action arg while ${e.phase} of your story.  
        ${e.deprecated ? `
This is deprecated and won't work in Storybook 8 anymore.
` : ""}
        Please provide an explicit spy to your args like this:
          import { fn } from '@storybook/test';
          ... 
          args: {
           ${e.name}: fn()
          }`
    });
    this.data = e;
  }
};
o(R, "ImplicitActionsDuringRendering");
var j = R, b = class b extends s {
  constructor() {
    super({
      category: "PREVIEW_API",
      code: 3,
      message: n`
        Cannot call \`storyStore.extract()\` without calling \`storyStore.cacheAllCsfFiles()\` first.

        You probably meant to call \`await preview.extract()\` which does the above for you.`
    });
  }
};
o(b, "CalledExtractOnStoreError");
var L = b, I = class I extends s {
  constructor() {
    super({
      category: "PREVIEW_API",
      code: 4,
      message: n`
        Expected your framework's preset to export a \`renderToCanvas\` field.

        Perhaps it needs to be upgraded for Storybook 7.0?`,
      documentation: "https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#mainjs-framework-field"
    });
  }
};
o(I, "MissingRenderToCanvasError");
var F = I, P = class P extends s {
  constructor(e) {
    super({
      category: "PREVIEW_API",
      code: 5,
      message: n`
        Called \`Preview.${e.methodName}()\` before initialization.
        
        The preview needs to load the story index before most methods can be called. If you want
        to call \`${e.methodName}\`, try \`await preview.initializationPromise;\` first.
        
        If you didn't call the above code, then likely it was called by an addon that needs to
        do the above.`
    });
    this.data = e;
  }
};
o(P, "CalledPreviewMethodBeforeInitializationError");
var G = P, x = class x extends s {
  constructor(e) {
    super({
      category: "PREVIEW_API",
      code: 6,
      message: n`
        Error fetching \`/index.json\`:
        
        ${e.text}

        If you are in development, this likely indicates a problem with your Storybook process,
        check the terminal for errors.

        If you are in a deployed Storybook, there may have been an issue deploying the full Storybook
        build.`
    });
    this.data = e;
  }
};
o(x, "StoryIndexFetchError");
var Y = x, S = class S extends s {
  constructor(e) {
    super({
      category: "PREVIEW_API",
      code: 7,
      message: n`
        Tried to render docs entry ${e.storyId} but it is a MDX file that has no CSF
        references, or autodocs for a CSF file that some doesn't refer to itself.
        
        This likely is an internal error in Storybook's indexing, or you've attached the
        \`attached-mdx\` tag to an MDX file that is not attached.`
    });
    this.data = e;
  }
};
o(S, "MdxFileWithNoCsfReferencesError");
var H = S, w = class w extends s {
  constructor() {
    super({
      category: "PREVIEW_API",
      code: 8,
      message: n`
        Couldn't find any stories in your Storybook.

        - Please check your stories field of your main.js config: does it match correctly?
        - Also check the browser console and terminal for error messages.`
    });
  }
};
o(w, "EmptyIndexError");
var K = w, k = class k extends s {
  constructor(e) {
    super({
      category: "PREVIEW_API",
      code: 9,
      message: n`
        Couldn't find story matching '${e.storySpecifier}'.

        - Are you sure a story with that id exists?
        - Please check your stories field of your main.js config.
        - Also check the browser console and terminal for error messages.`
    });
    this.data = e;
  }
};
o(k, "NoStoryMatchError");
var M = k, T = class T extends s {
  constructor(e) {
    super({
      category: "PREVIEW_API",
      code: 10,
      message: n`
        Couldn't find story matching id '${e.storyId}' after importing a CSF file.

        The file was indexed as if the story was there, but then after importing the file in the browser
        we didn't find the story. Possible reasons:
        - You are using a custom story indexer that is misbehaving.
        - You have a custom file loader that is removing or renaming exports.

        Please check your browser console and terminal for errors that may explain the issue.`
    });
    this.data = e;
  }
};
o(T, "MissingStoryFromCsfFileError");
var U = T, V = class V extends s {
  constructor() {
    super({
      category: "PREVIEW_API",
      code: 11,
      message: n`
        Cannot access the Story Store until the index is ready.

        It is not recommended to use methods directly on the Story Store anyway, in Storybook 9 we will
        remove access to the store entirely`
    });
  }
};
o(V, "StoryStoreAccessedBeforeInitializationError");
var X = V, _ = class _ extends s {
  constructor(e) {
    super({
      category: "PREVIEW_API",
      code: 12,
      message: n`
      
      To use mount in the play function, you must satisfy the following two requirements: 
      
      1. You *must* destructure the mount property from the \`context\` (the argument passed to your play function). 
         This makes sure that Storybook does not start rendering the story before the play function begins.
      
      2. Your Storybook framework or builder must be configured to transpile to ES2017 or newer. 
         This is because destructuring statements and async/await usages are otherwise transpiled away, 
         which prevents Storybook from recognizing your usage of \`mount\`.
      
      Note that Angular is not supported. As async/await is transpiled to support the zone.js polyfill. 
      
      More info: https://storybook.js.org/docs/writing-tests/interaction-testing#run-code-before-the-component-gets-rendered
      
      Received the following play function:
      ${e.playFunction}`
    });
    this.data = e;
  }
};
o(_, "MountMustBeDestructuredError");
var J = _, v = class v extends s {
  constructor() {
    super({
      category: "PREVIEW_API",
      code: 13,
      message: n`
        You must configure testingLibraryRender to use play in portable stories.
        
        import { render } from '@testing-library/[renderer]';
        
        setProjectAnnotations({
          testingLibraryRender: render,
        });
        
        For other testing renderers, you can configure \`renderToCanvas\` like so:
        
        import { render } from 'your-test-renderer';
        
        setProjectAnnotations({
          renderToCanvas: ({ storyFn }) => {
            const Story = storyFn();
            
            // Svelte
            render(Story.Component, Story.props);
            
            // Vue
            render(Story);
            
            // or for React
            render(<Story/>);
          },
        });`
    });
  }
};
o(v, "TestingLibraryMustBeConfiguredError");
var q = v, W = class W extends s {
  constructor(e) {
    super({
      category: "PREVIEW_API",
      code: 14,
      message: n`
        No render function available for storyId '${e.id}'
      `
    });
    this.data = e;
  }
};
o(W, "NoRenderFunctionError");
var z = W, A = class A extends s {
  constructor() {
    super({
      category: "PREVIEW_API",
      code: 15,
      message: n`
        No component is mounted in your story.
        
        This usually occurs when you destructure mount in the play function, but forget to call it.
        
        For example:

        async play({ mount, canvasElement }) {
          // 👈 mount should be called: await mount(); 
          const canvas = within(canvasElement);
          const button = await canvas.findByRole('button');
          await userEvent.click(button);
        };

        Make sure to either remove it or call mount in your play function.
      `
    });
  }
};
o(A, "NoStoryMountedError");
var Q = A, N = class N extends s {
  constructor() {
    super({
      category: "FRAMEWORK_NEXTJS",
      code: 1,
      documentation: "https://storybook.js.org/docs/get-started/nextjs#faq",
      message: n`
      You are importing avif images, but you don't have sharp installed.

      You have to install sharp in order to use image optimization features in Next.js.
      `
    });
  }
};
o(N, "NextJsSharpError");
var Z = N, $ = class $ extends s {
  constructor(e) {
    super({
      category: "FRAMEWORK_NEXTJS",
      code: 2,
      message: n`
        Tried to access router mocks from "${e.importType}" but they were not created yet. You might be running code in an unsupported environment.
      `
    });
    this.data = e;
  }
};
o($, "NextjsRouterMocksNotAvailable");
var B = $, C = class C extends s {
  constructor(e) {
    super({
      category: "DOCS-TOOLS",
      code: 1,
      documentation: "https://github.com/storybookjs/storybook/issues/26606",
      message: n`
        There was a failure when generating detailed ArgTypes in ${e.language} for:
        ${JSON.stringify(e.type, null, 2)} 
        
        Storybook will fall back to use a generic type description instead.

        This type is either not supported or it is a bug in the docgen generation in Storybook.
        If you think this is a bug, please detail it as much as possible in the Github issue.
      `
    });
    this.data = e;
  }
};
o(C, "UnknownArgTypesError");
var ee = C;
export {
  L as CalledExtractOnStoreError,
  G as CalledPreviewMethodBeforeInitializationError,
  se as Category,
  K as EmptyIndexError,
  j as ImplicitActionsDuringRendering,
  H as MdxFileWithNoCsfReferencesError,
  F as MissingRenderToCanvasError,
  D as MissingStoryAfterHmrError,
  U as MissingStoryFromCsfFileError,
  J as MountMustBeDestructuredError,
  Z as NextJsSharpError,
  B as NextjsRouterMocksNotAvailable,
  z as NoRenderFunctionError,
  M as NoStoryMatchError,
  Q as NoStoryMountedError,
  Y as StoryIndexFetchError,
  X as StoryStoreAccessedBeforeInitializationError,
  q as TestingLibraryMustBeConfiguredError,
  ee as UnknownArgTypesError
};
